src/System.Device.Gpio/System/Device/Pwm/PwmChannel.cs

```diff
@@ -67,14 +67,27 @@ protected virtual void Dispose(bool disposing)
                     frequency,
                     dutyCyclePercentage);
             }
-            else
+
+            if (IsBeagleBoneKernel())
             {
-                return new Channels.UnixPwmChannel(
-                    chip,
-                    channel,
-                    frequency,
-                    dutyCyclePercentage);
+                return new Channels.BeagleBonePwmChannel(
+                chip,
+                channel,
+                frequency,
+                dutyCyclePercentage);
             }
+
+            return new Channels.UnixPwmChannel(
+                chip,
+                channel,
+                frequency,
+                dutyCyclePercentage);
+        }
+
+        private static bool IsBeagleBoneKernel()
+        {
+            var kernelVersionInfo = System.IO.File.ReadAllText(@"/proc/version");
```

Comment:
NIT: can we add a try catch here in case /proc/version file can't be read for a permission issue and if so just return false?

Reply:
I wasn't sure if it would be appropriate to return false if it failed, but I'm happy to make this change if you feel like that's good behavior.

Reply:
what we want is to not regress existing scenarios and have this be kind of like "A nice addition when it's needed" kind of thing, so by surrounding this with a try catch it means that if for whatever reason it fails we always silently fall back to old behavior, but in normal cases it will just work and provide better experience.

Reply:
One other thing I forgot to mention, can you please switch the `var`'s for explicit types? as well as make the Contains to be InvariantCulture so that the comparison happens fast

Reply:
Sure. That makes sense.
I'll push an update up tomorrow morning.

All the `var`s? 
Is there a published style guide somewhere?
It would be nice to have some of these things in the [CONTRIBUTING.md](https://github.com/dotnet/iot/blob/master/Documentation/CONTRIBUTING.md) file.

Reply:
We have a link saying that we follow the same code styling than our other dotnet repos here: https://github.com/dotnet/iot/blob/master/Documentation/Coding-guidelines.md
If you go there and specifically check number 10, you'll see that:
> We only use var when it's obvious what the variable type is (e.g. var stream = new FileStream(...) not var stream = OpenStandardInput()).

Reply:
This is done.
I tried to document the trade offs being made with a comment.
Please let me know if it captured the intent well enough.